### PR TITLE
Give error if using instance_sizeof on a generic type without type vars

### LINKED ISSUE #5207: BUG: called llvm_struct_type for union of generics
```cr
class Foo(T)
  def initialize(@x : T)
  end
end

Foo(Int32).new(0)
Foo(Int64).new(0i64)
p instance_sizeof(Foo)
```

causes:

```
BUG: called llvm_struct_type for (Foo(Int32) | Foo(Int64))
.../src/compiler/crystal/codegen/llvm_typer.cr:410:7 in 'create_llvm_struct_type'
.../src/compiler/crystal/codegen/llvm_typer.cr:308:33 in 'llvm_struct_type'
.../src/compiler/crystal/codegen/llvm_typer.cr:302:5 in 'llvm_struct_type'
.../src/compiler/crystal/codegen/codegen.cr:83:26 in 'instance_size_of'
.../src/compiler/crystal/semantic/main_visitor.cr:2476:20 in 'visit'
.../src/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
.../src/compiler/crystal/semantic/main_visitor.cr:1236:21 in 'visit'
.../src/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
.../src/compiler/crystal/syntax/ast.cr:150:27 in 'accept_children'
.../src/compiler/crystal/syntax/visitor.cr:28:11 in 'accept'
.../src/compiler/crystal/semantic/main_visitor.cr:7:7 in 'visit_main'
.../src/compiler/crystal/semantic/main_visitor.cr:5:5 in 'visit_main:process_finished_hooks:cleanup'
.../src/compiler/crystal/semantic.cr:22:7 in 'semantic'
.../src/compiler/crystal/compiler.cr:141:7 in 'compile'
.../src/compiler/crystal/command.cr:265:7 in 'compile'
.../src/compiler/crystal/command.cr:184:14 in 'run_command'
.../src/compiler/crystal/command.cr:102:7 in 'run'
.../src/compiler/crystal/command.cr:49:5 in 'run'
.../src/compiler/crystal/command.cr:48:3 in 'run'
.../src/compiler/crystal.cr:8:1 in '__crystal_main'
.../src/crystal/main.cr:11:3 in '_crystal_main'
.../src/crystal/main.cr:112:5 in 'main_user_code'
.../src/crystal/main.cr:101:7 in 'main'
.../src/crystal/main.cr:135:3 in 'main'
__libc_start_main
_start
???
```

I don't see anything sane that could be returned from this `instance_sizeof`, so this should cause an error.

### PR DESCRIPTION
**Pull request by lbguilherme**
Fixes #5207

### CODE REVIEW

**Comment by sdogruyol** [comment]
LGTM :+1: ?

**Comment by RX14** [comment]
@sdogruyol in the interests of keeping noise down, can you not "LGTM" random prs, unless they need bumping.

**Comment by sdogruyol** [comment]
@RX14 they're mostly approved PRs not random..

**Comment by RX14** [comment]
Ok but its still not helpful

**Comment by asterite** [comment]
I agree, the LGTM end up in my mail box and when I think there's some new interesting reply or something relevant to the issue, it's just a 👍 . Let's avoid that.

File: src/compiler/crystal/semantic/main_visitor.cr [hunk 1]
```diff
@@ -2469,10 +2474,19 @@ module Crystal
       node.exp.accept self
       @in_type_args -= 1
 
+      type = node.exp.type?
+      unless type
+        raise "BUG: #{node} at #{node.location} should receive a type"
+      end
+
+      if type.is_a? GenericClassType
```
Comment (asterite) [inline src/compiler/crystal/semantic/main_visitor.cr]: This should be `GenericType`. That's the base module for both generic classes and modules.

**Comment by lbguilherme** [comment]
@asterite Done.

**Pull request merged**